node/pkg/guardiansigner/guardiansigner.go

```diff
@@ -0,0 +1,74 @@
+package guardiansigner
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"strings"
+)
+
+// The types of guardian signers that are supported
+type SignerType int
+
+const (
+	InvalidSignerType SignerType = iota
+	// file://<path-to-file>
+	FileSignerType
+)
+
+// GuardianSigner interface
+type GuardianSigner interface {
+	// Sign expects a keccak256 hash that needs to be signed.
+	Sign(hash []byte) (sig []byte, err error)
+	// PublicKey returns the ECDSA public key of the signer. Note that this should not
+	// be confused with the EVM address.
+	PublicKey() (pubKey ecdsa.PublicKey)
+	// Verify is a convenience function that recovers a public key from the sig/hash pair,
+	// and checks if the public key matches that of the guardian signer.
+	Verify(sig []byte, hash []byte) (valid bool, err error)
+}
+
+func NewGuardianSignerFromUri(signerUri string, unsafeDevMode bool) (GuardianSigner, error) {
+
+	// Get the signer type
+	signerType, signerKeyConfig := ParseSignerUri(signerUri)
+
+	switch signerType {
+	case FileSignerType:
+		return NewFileSigner(unsafeDevMode, signerKeyConfig)
+	default:
+		return nil, fmt.Errorf("unsupported guardian signer type")
+	}
+}
+
+func ParseSignerUri(signerUri string) (signerType SignerType, signerKeyConfig string) {
+	// Split the URI using the standard "://" scheme separator
+	signerUriSplit := strings.Split(signerUri, "://")
+
+	// Length smaller than 2 implies that there is no path following the scheme separator,
+	// which is invalid.
+	if len(signerUriSplit) < 2 {
```

Comment:
If there was no path following the scheme separator won't we still get 2 elements, but the second is just an empty string?

Reply:
will update comment that we are checking for a separator, and we don't care about the rest.